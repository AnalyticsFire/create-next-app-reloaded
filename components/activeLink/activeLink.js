import React from 'react';
import classnames from 'classnames';
import { withRouter } from 'next/router';
import { string, object, any } from 'prop-types';

// typically you want to use `next/link` for this usecase
// but this example shows how you can also access the router
// using the withRouter utility.

export const ActiveLink = ({ children, router, href, className }) => {
  const handleClick = e => {
    e.preventDefault();
    router.push(href);
  };

  return (
    <a
      href={href}
      onClick={handleClick}
      className={classnames(className, { active: router.pathname === href })}
    >
      {children}
    </a>
  );
};

ActiveLink.propTypes = {
  children: any,
  router: object,
  href: string,
  className: string,
};

export default withRouter(ActiveLink);
