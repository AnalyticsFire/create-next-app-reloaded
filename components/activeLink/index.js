import Component, { ActiveLink } from './activeLink';

export { ActiveLink };
export default Component;
