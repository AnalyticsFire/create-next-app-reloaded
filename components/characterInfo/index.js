import Component, { CharacterInfo } from './characterInfo';

export { CharacterInfo };
export default Component;
