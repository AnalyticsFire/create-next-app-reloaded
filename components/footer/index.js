import Component, { Footer } from './footer';

export { Footer };
export default Component;
