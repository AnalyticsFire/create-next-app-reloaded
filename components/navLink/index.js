import NavLink from './navLink';

export default NavLink;
