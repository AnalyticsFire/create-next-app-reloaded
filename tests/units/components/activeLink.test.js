/* eslint-disable no-undef */

import React from 'react';
import toJson from 'enzyme-to-json';
import { render } from 'enzyme';
import { ActiveLink } from '@root/components/activeLink';

const defaultComponent = (
  <ActiveLink router={{ pathname: '/home' }} href="/home" className="link">
    Home
  </ActiveLink>
);

test('ActiveLink is rendered', () => {
  expect(toJson(render(defaultComponent))).toMatchSnapshot();
});

/* eslint-enable no-undef */
