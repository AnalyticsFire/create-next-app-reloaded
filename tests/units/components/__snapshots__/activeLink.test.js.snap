// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveLink is rendered 1`] = `
<a
  class="link active"
  href="/home"
>
  Home
</a>
`;
